src/icmp/Icmp4.cc

```diff
@@ -156,7 +156,7 @@ void
 Icmp4::Recv(void)
 {
     int n;
-    struct addrinfo *from = nullptr;
+    Ip::Address from;
```

Comment:
`from` should be of type `sockaddr_storage` and empty.
```suggestion
    sockaddr_storage from;
    memset(&from, 0, sizeof(from));
```

Also, these lines should be down just before first-use, where the `recvfrom()` is called.